cumulus/parachains/runtimes/assets/asset-hub-rococo/src/lib.rs

```diff
@@ -1721,12 +1723,17 @@ impl_runtime_apis! {
 			build_state::<RuntimeGenesisConfig>(config)
 		}
 
-		fn get_preset(id: &Option<sp_genesis_builder::PresetId>) -> Option<Vec<u8>> {
-			get_preset::<RuntimeGenesisConfig>(id, |_| None)
+		fn get_preset(id: &Option<PresetId>) -> Option<Vec<u8>> {
+			get_preset::<RuntimeGenesisConfig>(id, &genesis_config_presets::get_preset)
 		}
 
-		fn preset_names() -> Vec<sp_genesis_builder::PresetId> {
-			vec![]
+		fn preset_names() -> Vec<PresetId> {
+			use crate::genesis_config_presets::preset_names::*;
+			vec![
+				PresetId::from(PRESET_GENESIS),
+				PresetId::from(PRESET_DEVELOPMENT),
+				PresetId::from(PRESET_LOCAL),
```

Comment:
They got be directly `PresetId`, so you don't need to convert them. Maybe adding some `const fn new(&'static str)` would be a good idea.

Reply:
I spent some time on this, but failed. PresetId is an alias for RuntimeString, things are getting complicated when I try to use it in match. I can open follow-up issue to clean this up.

Reply:
-> https://github.com/paritytech/polkadot-sdk/issues/5532